src/main/java/com/woodencloset/signalbot/Main.java

```diff
@@ -45,7 +49,12 @@ public static void main(String[] args) throws IOException, InvalidKeyException,
         bot.addResponder(new HebrewDiceRollResponder());
 
         if (cmd.hasOption("register-text")) {
-            bot.register(cmd.getOptionValue("register-text"), SignalBot.RegistrationType.TextMessage);
```

Comment:
Instead of this, I would just set `String captcha = cmd.hasOption(...) ? cmd.getOptionValue(...) : null;` and then pass it as the third argument to `bot.register`. This would then work if you add it as an argument to `register` and use `Optional.ofNullable(captcha)` when calling `requestSmsVerificationCode` or `requestVoiceVerificationCode`